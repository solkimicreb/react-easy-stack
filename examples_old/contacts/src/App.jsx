import React from "react";
import { view } from "react-easy-state";
import Contact from "./Contact";
import ContactCreator from "./ContactCreator";
import appStore from "./appStore";

// this rerenders whenever the appStore.contacts array changes (elements pushed or deleted)
function App() {
  return (
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Email</th>
          <th>Action</th>
        </tr>
      </thead>
      <tbody>
        {appStore.contacts.map(contact => (
          <Contact contact={contact} key={contact.email} />
        ))}
        <ContactCreator />
      </tbody>
    </table>
  );
}

// wrap the component with view() before exporting it
export default view(App);
