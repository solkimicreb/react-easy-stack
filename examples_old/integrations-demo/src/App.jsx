import React from "react";
import { view } from "react-easy-state";
import { params, path } from "react-easy-params";

const updateParam = ev => (params.name = ev.target.value);
const updatePath = ev => (path[0] = ev.target.value);

export default view(() => (
  <div>
    <div>
      Path: <input onChange={updatePath} value={path[0]} />
    </div>
    <div>
      Param: <input onChange={updateParam} value={params.name} />
    </div>
  </div>
));
