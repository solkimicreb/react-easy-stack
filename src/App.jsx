import React, { Component } from 'react'
import Router from './Router'
import Dummy from './Dummy'
import Link from './Link'

export default class App extends Component {
  async onRoute () {
    await new Promise(resolve => setTimeout(resolve, 1))
  }

  render () {
    return (
      <div>
        <Link to='/hello' element='li'>
          Link to Hello
        </Link>
        <Link to='/hi' element='li' params={{ name: 'Berci' }}>
          Link to Hi
        </Link>

        <Router>
          <Dummy page='hello' />
          <Router page='hi' default onRoute={this.onRoute}>
            <Dummy page='world' default />
            <Dummy page='you' />
          </Router>
        </Router>

        <Router>
          <Dummy page='hello' />
          <Dummy page='hi' default />
        </Router>
      </div>
    )
  }
}
