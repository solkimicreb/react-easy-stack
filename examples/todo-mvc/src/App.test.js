import React from "react";
import enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import { Link } from "react-easy-stack";
import App from "./App";

enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("TodoMVC App", () => {
  const app = mount(<App />);

  test("should add todos", () => {
    expect(app).toMatchSnapshot("01. Initial state");

    const input = app.find(".new-todo");

    input.simulate("keyUp", {
      keyCode: 13,
      target: { value: "Test Todo" }
    });
    expect(app).toMatchSnapshot("02. Add Test Todo");

    input.simulate("keyUp", {
      keyCode: 13,
      target: { value: "Other Todo" }
    });
    expect(app).toMatchSnapshot("03. Add Other Todo");

    input.simulate("keyUp", {
      keyCode: 27,
      target: { value: "Final Tod" }
    });
    input.simulate("keyUp", {
      keyCode: 13,
      target: { value: "Final Todo" }
    });
    expect(app).toMatchSnapshot("04. Add Final Todo");
  });

  test("should toggle todo status", () => {
    const toggles = app.find(".todo-list .toggle");

    toggles.at(0).simulate("change");
    expect(app).toMatchSnapshot("05. Toggle Test Todo to completed");

    toggles.at(1).simulate("change");
    expect(app).toMatchSnapshot("06. Toggle Other Todo to completed");

    toggles.at(0).simulate("change");
    expect(app).toMatchSnapshot("07. Toggle Test Todo to active");
  });

  test("should filter todos", () => {
    const allFilter = app.find(Link).at(0);
    const activeFilter = app.find(Link).at(1);
    const completedFilter = app.find(Link).at(2);

    completedFilter.simulate("click");
    expect(app).toMatchSnapshot("08. Filter completed");

    activeFilter.simulate("click");
    expect(app).toMatchSnapshot("09. Filter active");

    allFilter.simulate("click");
    expect(app).toMatchSnapshot("10. Filter all");
  });

  test("should clear completed", () => {
    const clearCompleted = app.find(".clear-completed");

    clearCompleted.simulate("click");
    expect(app).toMatchSnapshot("11. Clear completed");
  });

  test("should toggle all todo state at once", () => {
    const toggleAll = app.find(".toggle-all");

    toggleAll.simulate("change");
    expect(app).toMatchSnapshot("12. Toggle all to completed");

    toggleAll.simulate("change");
    expect(app).toMatchSnapshot("13. Toggle all to active");
  });

  test("should delete todo", () => {
    const deleters = app.find(".todo-list .destroy");

    deleters.at(0).simulate("click");
    expect(app).toMatchSnapshot("14. Delete Test Todo");
  });
});
