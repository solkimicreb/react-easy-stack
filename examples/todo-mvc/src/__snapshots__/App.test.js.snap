// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoMVC App should add todos: 01. Initial state 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
</div>
`;

exports[`TodoMVC App should add todos: 02. Add Test Todo 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      1
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
  </footer>
</div>
`;

exports[`TodoMVC App should add todos: 03. Add Other Todo 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
  </footer>
</div>
`;

exports[`TodoMVC App should add todos: 04. Add Final Todo 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      3
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
  </footer>
</div>
`;

exports[`TodoMVC App should clear completed: 11. Clear completed 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
  </footer>
</div>
`;

exports[`TodoMVC App should delete todo: 14. Delete Test Todo 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      1
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
  </footer>
</div>
`;

exports[`TodoMVC App should filter todos: 08. Filter completed 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;

exports[`TodoMVC App should filter todos: 09. Filter active 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;

exports[`TodoMVC App should filter todos: 10. Filter all 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;

exports[`TodoMVC App should toggle all todo state at once: 12. Toggle all to completed 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={true}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      0
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;

exports[`TodoMVC App should toggle all todo state at once: 13. Toggle all to active 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
  </footer>
</div>
`;

exports[`TodoMVC App should toggle todo status: 05. Toggle Test Todo to completed 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;

exports[`TodoMVC App should toggle todo status: 06. Toggle Other Todo to completed 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      1
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;

exports[`TodoMVC App should toggle todo status: 07. Toggle Test Todo to active 1`] = `
<div
  className="todoapp"
>
  <header
    className="header"
  >
    <h1>
      todos
    </h1>
    <input
      autoFocus={true}
      className="new-todo"
      onKeyUp={[Function]}
      placeholder="What needs to be done?"
    />
  </header>
  <section
    className="main"
  >
    <input
      checked={false}
      className="toggle-all"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="toggle-all"
    >
      Mark all as complete
    </label>
    <ul
      className="todo-list"
    >
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Test Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view completed"
      >
        <input
          checked={true}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Other Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
      <li
        className="view"
      >
        <input
          checked={false}
          className="toggle"
          onChange={[Function]}
          type="checkbox"
        />
        <label>
          Final Todo
        </label>
        <button
          className="destroy"
          onClick={[Function]}
        />
      </li>
    </ul>
  </section>
  <footer
    className="footer"
  >
    <span
      className="todo-count"
    >
      2
       items left
    </span>
    <div
      className="filters"
    >
      <a
        className=" selected"
        href="/blank?filter=%22all%22"
        onClick={[Function]}
        style={Object {}}
      >
        All
      </a>
      <a
        className=""
        href="/blank?filter=%22active%22"
        onClick={[Function]}
        style={Object {}}
      >
        Active
      </a>
      <a
        className=""
        href="/blank?filter=%22completed%22"
        onClick={[Function]}
        style={Object {}}
      >
        Completed
      </a>
    </div>
    <button
      className="clear-completed"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</div>
`;
